### PR TITLE
feature: binlog to external or naglfar cluster

### LINKED ISSUE #73: feature: support binlog
## Development Task
We need support binlog scene. As follows:
1.I created tct,tct2. I want to set tct binlog to tct2.
2.I created tct. I want to set tct binlog to external tidb.

### PR DESCRIPTION
**Pull request by CadmusJiang**
### What problem does this PR solve?

Issue Number: close #73  

Problem Summary: support binlog

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
1.add pump and drainer config in testclustertopology_types.go
2.transfer syncer.to.host in deploy.go
How it Works:
In deploy.go check syncer.to.host if have "naglfar://", if have tansfer "naglfar://n1" to "172.16.0.2"(clusterIP), other do nothing.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
1.apply machine.yaml
2.apply request.yaml(include n1,n2)
3.apply topology2.yaml(create topology in n2)
4.apply topology.yaml(create topology in n1, binlog to n2)

topology.yaml
```yaml
apiVersion: naglfar.pingcap.com/v1
kind: TestClusterTopology
metadata:
  name: tct
  namespace: test
spec:
  resourceRequest: tr
  tidbCluster:
    serverConfigs:
      tidb: |-
        log.slow-threshold: 200
        binlog.enable: true
        binlog.ignore-error: true
  ...
    pump:
      - host: n1
        sshPort: 22
        port: 8250
        deployDir: "/tidb-deploy/pump-8249"
        dataDir: "/tidb-data/pump-8249"
        config: |-
          gc: 7
    drainer:
      - host: n1
        sshPort: 22
        port: 8249
        dataDir: "/tidb-data/drainer-8249"
        commitTS: -1
        deployDir: "/tidb-deploy/drainer-8249"
        syncerKind: "internal"
        config: |-
          syncer.db-type: "tidb"
          syncer.to.host: "naglfar://n2"
          syncer.to.user: "root"
          syncer.to.password: ""
          syncer.to.port: 4000
...

```
Side effects

- Breaking backward compatibility

### CODE REVIEW

File: pkg/tiup/deploy.go [hunk 1]
```diff
@@ -63,6 +63,60 @@ func IgnoreClusterNotExist(err error) error {
...
+		drainerConfig,
+	}} {
+		err := unmarshalDrainerConfigToMap([]byte(item.config), item.object)
+		if err != nil {
+			return err
+		}
+	}
+	// transfer
+	if clusterIPMaps != nil {
+		if strings.HasPrefix(config["syncer.to.host"].(string), "naglfar://") {
```
Comment (mahjonp) [inline pkg/tiup/deploy.go:112]: parsing and transforming protocol of `syncer.to.host` should put before `if clusterIPMaps != nil`, and you'd better ensure that the naglfar node exists in `clusterIPMaps`.

**Review by mahjonp — COMMENTED** [review]
Rest LGTM

File: pkg/tiup/deploy.go (hunk 1 above)
Comment (CadmusJiang) [reply pkg/tiup/deploy.go:112]: Install, update, delete will use MakeClusterManager(BuildSpecification), I think this scenes and other maybe all need drainers.Some scenes don't need parse "naglfar://n1", but "naglfar://n1" must be parsed correctly when clusterIPMaps is not empty.I will change it as follow.
```go
// transfer
	if clusterIPMaps != nil {
		clusterIP := clusterIPMaps[0][strings.Split(config["syncer.to.host"].(string), "://")[1]]
		if clusterIP == "" {
			return fmt.Errorf("syncer.to.host is not existed")
		} else {
			config["syncer.to.host"] = clusterIP
		}
	}
```

**Review by mahjonp — APPROVED** [review]
lgtm

**Pull request merged**